plugins/flytekit-pydantic/flytekitplugins/pydantic/serialization.py

```diff
@@ -0,0 +1,128 @@
+"""
+Logic for serializing a basemodel to a literalmap that can be passed between container
+
+The serialization process is as follows:
+
+1. Serialize the basemodel to json, replacing all flyte types with unique placeholder strings
+2. Serialize the flyte types to separate literals and store them in the flyte object store (a singleton object)
+3. Return a literal map with the json and the flyte object store represented as a literalmap {placeholder: flyte type}
+
+"""
+from typing import Any, Dict, NamedTuple, Union, cast
+from typing_extensions import Annotated
+import uuid
+
+import pydantic
+from google.protobuf import struct_pb2
+
+from flytekit.models import literals
+from flytekit.core import context_manager, type_engine
+
+from . import commons
+
+
+BASEMODEL_JSON_KEY = "BaseModel JSON"
+FLYTETYPE_OBJSTORE__KEY = "Flyte Object Store"
+
+SerializedBaseModel = Annotated[str, "A pydantic BaseModel that has been serialized with placeholders for Flyte types."]
```

Comment:
Yeah let's discuss that. I think it could work. Need to have a look at how the dict() method on the BaseModel works in the context of this integration.

Reply:
Gave my long comment at the bottom so marking this note as resolved